src/plugins/content_management/public/registry/content_type.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ContentTypeDetails, ContentTypeKind } from './content_type_details';
+import type { CrudClient } from '../crud_client';
+
+export class ContentType {
```

Comment:
The naming do not work well with what is server side. I know naming is hard but can we try to find common terminology?

For me contentType is.. a content `type`. Meaning "dashboard", "map"... It correspond (in most cases) to a Saved object `type`.

So using here `ContentType` for an interface is strange.

Why not simply call this interface `Content` ?

Then instead of `id`, we would have `type`. This would correspond to all the `type|contentType` we pass to the rpc layer and crud instances server side.

WDYT?